### PR TITLE
Fix the selectbox celltype behavior when multiselected 

### PR DESCRIPTION
**Pull request by parleur**
PR on top of @parente 's #809 (ping @Carreau, @jdfreder )
Basically when several cells are selected, the selectbox enter in a state '-' hidden otherwise. It allows to select any other cell type and flip all cell in selection to it. 

Thanks to @carreau for help on it.

### CODE REVIEW

File: notebook/static/notebook/js/maintoolbar.js [hunk 1]
```diff
@@ -89,19 +89,27 @@ define([
...
+            .append($('<option/>').attr('value','heading').text('Heading'))
+            .append(multiselect);
         this.notebook.keyboard_manager.register_events(sel);
         this.events.on('selected_cell_type_changed.Notebook', function (event, data) {
-            if (data.cell_type === 'heading') {
-                sel.val('Markdown');
+            if ( that.notebook.get_selected_cells_indices().length > 1) {
+                multiselect.show();
+                sel.val('multiselect');
+                multiselect.hide();
```
Comment (Carreau) [inline notebook/static/notebook/js/maintoolbar.js]: I think you want the `multiselect.hide()` in the else part of the branch.

File: notebook/static/notebook/js/maintoolbar.js [hunk 2]
```diff
@@ -89,19 +89,27 @@ define([
...
             } else {
-                sel.val(data.cell_type);
+                if (data.cell_type === 'heading') {
```
Comment (jdfreder) [inline notebook/static/notebook/js/maintoolbar.js]: @Carreau can correct me if I'm wrong, but I don't think this cell type exists anymore.

Comment (Carreau) [inline notebook/static/notebook/js/maintoolbar.js]: This is just indented from before, heading technically still exists with keyboard shortcut, it switched to markdown and insert the correct amount of `#`

Comment (jdfreder) [inline notebook/static/notebook/js/maintoolbar.js]: Got it

File: notebook/static/notebook/js/maintoolbar.js (hunk 1 above)
Comment (parleur) [inline notebook/static/notebook/js/maintoolbar.js]: Whaou. I am surprised it works.  You can programatically select a hidden option, it is shown and back hidden when unselected... good to known.

File: notebook/static/notebook/js/maintoolbar.js [hunk 3]
```diff
@@ -89,19 +89,26 @@ define([
     // encountered when building a toolbar.
     MainToolBar.prototype._pseudo_actions.add_celltype_list = function () {
         var that = this;
+        var multiselect = $('<option/>').attr('value','multiselect').attr('disabled','').text('-');
+        multiselect.hide();
```
Comment (Carreau) [inline notebook/static/notebook/js/maintoolbar.js]: No, not here. Or i is never visible.

File: notebook/static/notebook/js/maintoolbar.js [hunk 4]
```diff
@@ -89,19 +89,26 @@ define([
...
             .append($('<option/>').attr('value','raw').text('Raw NBConvert'))
-            .append($('<option/>').attr('value','heading').text('Heading'));
+            .append($('<option/>').attr('value','heading').text('Heading'))
+            .append(multiselect);
         this.notebook.keyboard_manager.register_events(sel);
         this.events.on('selected_cell_type_changed.Notebook', function (event, data) {
-            if (data.cell_type === 'heading') {
-                sel.val('Markdown');
+            if ( that.notebook.get_selected_cells_indices().length > 1) {
+                sel.val('multiselect');
```
Comment (Carreau) [inline notebook/static/notebook/js/maintoolbar.js]: put show() line 106,

File: notebook/static/notebook/js/maintoolbar.js [hunk 5]
```diff
@@ -89,19 +89,26 @@ define([
...
             } else {
-                sel.val(data.cell_type);
+                if (data.cell_type === 'heading') {
```
Comment (Carreau) [inline notebook/static/notebook/js/maintoolbar.js]: and hide() somwhere in this block :-)

**Comment by Carreau** [comment]
Perfect thanks, we can refine later to have a better logic that does the same things only if all the selected cells do not have the same type !

**Pull request merged**